terraform/files/template/cluster-template.yaml

```diff
@@ -83,17 +83,65 @@ spec:
         permissions: "0755"
         content: |
           #!/bin/bash
+          # Defragment & backup & trim script for SCS k8s-cluster-api-provider etcd cluster
+          # Script exits without any defrag/backup/trim action if:
+          #  - It is executed on non leader etcd member
+          #  - It is executed on single member etcd cluster
+          # Script defragment the etcd cluster as follows:
+          #  - Defrag the non leader etcd members first
+          #  - Change the leadership to the randomly selected and defragmentation completed etcd member
+          #  - Defrag the local (ex-leader) etcd member
+          # Script then backup & trim local (ex-leader) etcd member
+          #
+          # Usage: etcd-defrag.sh
+
           export LOG_DIR=/var/log
           export ETCDCTL_API=3
-          if test "$(etcdctl --cert /etc/kubernetes/pki/etcd/peer.crt --key /etc/kubernetes/pki/etcd/peer.key --cacert /etc/kubernetes/pki/etcd/ca.crt endpoint status | cut -d ',' -f 5)" != " false"; then
-              echo "Exit on leader"
-              exit 0
+          ETCDCTL="etcdctl --cert /etc/kubernetes/pki/etcd/peer.crt --key /etc/kubernetes/pki/etcd/peer.key --cacert /etc/kubernetes/pki/etcd/ca.crt"
+
+          if test "$($ETCDCTL endpoint status | cut -d ',' -f 5 | tr -d [:blank:])" = "false"; then
+            echo "Exit on non leader"
+            exit 0
+          fi
+
+          # Get all etcd members with their endpoints, IDs, and leader status
+          declare -a MEMBERS
+          while read MEMBER; do
+            MEMBERS+=( "$MEMBER" )
+          done < <($ETCDCTL endpoint status --cluster)
+
+          if test $${#MEMBERS[@]} = 1; then
```

Comment:
Note: $${#MEMBERS[@]} - double `$` here and below as a escape character for `clusterctl generate` command